src/components/component.ts

```diff
@@ -450,10 +450,10 @@ export class Component {
 
     if (this._isAnchored) {
       this._element.remove();
+      this._isAnchored = false;
+      this._onDetachCallbacks.callCallbacks(this);
```

Comment:
Main reason why this is here is because I was playing with things to fix this. This change makes sense to me, why would we call the callbacks if not detaching is performed. 

I agree that the change does not make sense in the context of this PR, so I am happy to drop it if you have a strong preference towards me doing so. 

However, I think the change is needed and it's an overhead to open a new PR